src/prettyprint.lua

```diff
@@ -0,0 +1,229 @@
+local COLOURS = require("src.utils.consoleColor")
+local PASS_SYMBOL = COLOURS.PASS("√")
+local ERROR_SYMBOL = "●"
+local FAIL_SYMBOL = COLOURS.FAIL("×")
+local PASS_HEADER = COLOURS.PASS_HEADER(" PASS ")
+local FAIL_HEADER = COLOURS.FAIL_HEADER(" FAIL ")
+
+-- Matches Jest
+local TAB_SIZE = 2
+
+local NodeType = require("src.interface.testnodetype")
+local tablex = require("src.utils.tablex")
+local prettyValue = require("src.utils.prettyValue")
+
+local function getTabs(amount)
+	return (" "):rep(amount * TAB_SIZE)
+end
+
+--- Recursively traverses the node tree while calling the supplied callback.
+---@param node lest.TestResult|lest.Describe
+---@param onNode fun(node: lest.TestResult|lest.Describe, depth: number):boolean A callback which is called for each Describe and Test in the tree. You can return true to keep going or return false to terminate.
+---@param onEnd fun()|nil A callback which is called once a branch in the tree has finished being traversed.
+local function traverseNodes(node, onNode, onEnd, _depth)
+	_depth = _depth or 0
```

Comment:
why is this marked with an underscore, and why isn't it in the annotations?

Reply:
It's an internal depth counter only exposed to the `onNode` callback. Used for the tree view.

Reply:
`_` indicates an unused parameter.

Even if it's internal it shouldn't be hidden from annotations.

If you actually want to prevent calling the inner method (which you don't need to as this is an internal method), then you can declare a function within this that contains the actual logic. This is used for the public `runTests` method.